Sources/OktaIdx/IDXFormField.swift

```diff
@@ -0,0 +1,138 @@
+            get { _value }
+            set {
+                guard isMutable else { return }
+                _value = newValue
+            }
+        }
+        
+        /// Indicates whether or not the form value is read-only.
+        @objc public let isMutable: Bool
+        
+        /// Indicates whether or not the form value is required to successfully proceed through this remediation option.
+        @objc public let isRequired: Bool
+        
+        /// Indicates whether or not the value supplied in this form value should be considered secret, and not presented to the user.
+        @objc public let isSecret: Bool
+        
+        /// For composite form fields, this contains the nested array of form values to group together.
+        @objc public let form: IDXClient.Remediation.Form?
+        
+        /// For form fields that have specific options the user can choose from (e.g. security question, passcode, etc), this indicates the different form options that should be displayed to the user.
+        @objc public let options: [IDXClient.Remediation.Form.Field]?
+        
+        /// Indicates if this field is the selected option within a parent field's `options` array.
+        @objc public internal(set) var isSelectedOption: Bool
+        
+        /// Allows a developer to set the selected option for a field that contains multiple `options`.
+        ///
+        /// This will update the `isSelectedOption` on all relevant fields.
+        @objc public weak var selectedOption: IDXClient.Remediation.Form.Field? {
+            didSet {
+                guard let options = options else { return }
+                for option in options {
+                    option.isSelectedOption = (option === selectedOption)
+                }
+            }
+        }
+        
+        /// The list of messages sent from the server.
+        ///
+        /// Messages reported from the server at the FormValue level should be considered relevant to the individual form field, and as a result should be displayed to the user alongside any UI elements associated with it.
+        @objc public let messages: IDXClient.MessageCollection
+        
+        /// Relates this field to an authenticator, when a field is used to represent an authenticator. For example, when a field is used within a series of `options` to identify which authenticator to select.
+        @objc public internal(set) weak var authenticator: IDXClient.Authenticator?
+
+        /// Returns the nested `form` field with the given name.
+        @objc public subscript(name: String) -> Field? {
+            form?[name]
+        }
+        
+        let isVisible: Bool
+        let relatesTo: String?
+        var _value: AnyObject?
+        lazy var hasVisibleFields: Bool = {
+            if isVisible {
+                return true
+            }
+            
+            if let form = form,
+               !form.fields.lazy.filter({ $0.hasVisibleFields }).isEmpty
```

Comment:
I guess `.first(where: { $0.hasVisibleFields }) != nil` is better for performance. Of course here `lazy` solves performance issue. 
But take a look at `line 99`.

Reply:
Yeah, I was trying to allow it to abort early to improve performance; good catch on line 99, I'll include that in another PR.